code/modules/supermatter/supermatter.dm

```diff
@@ -278,6 +279,10 @@
 		if (!(l in oview(rad_range, src)) && !(l in range(src, round(rad_range * 2/3))))
 			continue
 		l.apply_effect(rads, IRRADIATE, blocked = l.getarmor(null, "rad"))
+		if(l.is_diona())
+			l.adjustToxLoss(-rads)
+			if(last_taste_time + 800 < world.time) // Not to spam message
+				to_chat(l, "<span class='notice'>You can feel an extreme level of energy which flows throught your body which makes you regenerate very fast.</span>")
```

Comment:
I would suggest changing the wording of that to the following so we can avoid the double which:

> You can feel an extreme level of energy which flows throught your body and makes you regenerate very fast.